facade/src/main/java/org/jboss/pnc/facade/providers/ArtifactProviderImpl.java

```diff
@@ -288,6 +321,41 @@ public ArtifactRevision getRevision(String id, Integer rev) {
         return artifactRevisionMapper.toDTO(auditedArtifact);
     }
 
+    private CriteriaQuery<Tuple> artifactInfoQuery(
+            CriteriaBuilder cb,
+            Optional<String> identifierPattern,
+            Optional<Set<ArtifactQuality>> qualities,
+            Optional<RepositoryType> repoType) {
+
+        CriteriaQuery<Tuple> query = cb.createTupleQuery();
+
+        Root<Artifact> artifact = query.from(org.jboss.pnc.model.Artifact.class);
+        query.multiselect(
+                artifact.get(Artifact_.id),
+                artifact.get(Artifact_.identifier),
+                artifact.get(Artifact_.artifactQuality));
+        Predicate withIdentifierLike = identifierPattern.isPresent()
+                ? cb.like(artifact.get(Artifact_.identifier), "%" + identifierPattern.get().replace("*", "%") + "%")
```

Comment:
Is there a particular reason to surround the identifier with '%'? 

If you'd asked for "org.jboss.pnc:pnc:jar:2.0.0.GA-redhat-00001" specifically. You could potentially get 3 artifacts (or more): 
- org.jboss.pnc:pnc:jar:2.0.0.GA-redhat-00001
- org.jboss.pnc:pnc:jar:2.0.0.GA-redhat-00001:sources
- org.jboss.pnc:pnc:jar:2.0.0.GA-redhat-00001:javadocs

Reply:
Well I thought that if the endpoint input is the identifier "pattern" then it should behave as LIKE functionality all the time. But if the case is more that it will be used for specific version calls then it should be probably based only on "*" present in the pattern.

Reply:
The queries for the identifier MUST NEVER have "%" at the begining, so we can utilize index prefix search in DB.

Reply:
OK, so I will leave only the 'replace' part and DA will specify by "*" whenever it will need to search all identifiers that match the pattern.